scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScaladocParser.scala

```diff
@@ -49,6 +49,9 @@ object ScaladocParser {
   private def codePrefix[_: P] = P("{{{")
   private def codeSuffix[_: P] = P(hspaces0 ~ "}}}")
 
+  def markdownCodePrefix[_: P] = P(leadHspaces0 ~ "```scala" ~ (!nl ~ AnyChar).rep ~ nl)
```

Comment:
is scala required? does dotty support only a subset of markdown, and a simple triple backquote without scala not supported?

Reply:
As far as I can see, syntax highlighting in scaladoc is supported for both triple backquotes with and without scala. However, code block with scala is presented a little bit differently (it's surrounded by a frame), and a new snippet compilation feature supports only ```scala. Also, I think scalafmt should try to format only scala code.

Reply:
> As far as I can see, syntax highlighting in scaladoc is supported for both triple backquotes with and without scala. However, code block with scala is presented a little bit differently (it's surrounded by a frame), and a new snippet compilation feature supports only ```scala. Also, I think scalafmt should try to format only scala code.

scalafmt will format the scaladoc. if you don't call out `` ```java ... ``` `` code as a code block, this part will simply be turned into simple text and wrapped into a paragraph, losing structure.

if you additionally want to teach scalafmt to format the scala code within scaladoc, that would be a separate task -- and i am not sure i know off the top of my head how to do it.

Reply:
> if you additionally want to teach scalafmt to format the scala code within scaladoc, that would be a separate task -- and i am not sure i know off the top of my head how to do it.

That was actually the goal.

Reply:
@kitbellew I created an issue for it: https://github.com/scalameta/scalafmt/issues/2498